relationship's order_by doesn't allow multiple clauses

**Describe the bug**

The typing of `relationship`'s `order_by` parameter doesn't allow passing multiple 'order by' clauses.

**Expected behavior**

The typing should reflect the runtime behaviour, which allows passing multiple clauses to order the relationship.

I noticed other `order_by` parameters/functions seems to just take `Any` as the clauses, but potentially all could be updated to something like:

```python
Union[Literal[False], str, Column, Callable[[], Column], List[Union[str, Column]]]
```
(with or without the `Literal[False]`, as appropriate.)

**To Reproduce**

```py
from sqlalchemy import Column, Integer, ForeignKey
from sqlalchemy.orm import relationship, declarative_base
from typing import List, Optional

Base = declarative_base()


class Foo(Base):
    __tablename__ = "foo"
    id = Column(Integer(), primary_key=True)
    bar_id: Optional[int] = Column(ForeignKey("bar.id"))

    a = Column(Integer())
    b = Column(Integer())


class Bar(Base):
    __tablename__ = "bar"
    id = Column(Integer(), primary_key=True)

    foos: List[Foo] = relationship(Foo, uselist=True, order_by=[Foo.a, Foo.b])
```

**Error**

```
order_by.py:21: error: Argument "order_by" to "RelationshipProperty" has incompatible type "List[Mapped[Optional[int]]]"; expected "Union[Literal[False], str, Column[Any], Callable[[], Column[Any]]]"
Found 1 error in 1 file (checked 1 source file)
```

**Versions.**
 - OS: n/a
 - Python: 3.8.7
 - SQLAlchemy: see requirements below
 - Database: n/a
 - DBAPI: n/a

**Additional context**

requirements:

```
greenlet==1.0.0
mypy==0.812
mypy-extensions==0.4.3
SQLAlchemy==1.4.11
sqlalchemy2-stubs==0.0.1a4
typed-ast==1.4.3
typing-extensions==3.7.4.3
```

--- a/sqlalchemy-stubs/orm/relationships.pyi
+++ b/sqlalchemy-stubs/orm/relationships.pyi
@@ -27,8 +27,8 @@ from .. import schema as schema
 from .. import sql as sql
 from .. import util as util
 from ..inspection import inspect as inspect
-from ..schema import Column
 from ..sql import coercions as coercions
+from ..sql import elements as elements
 from ..sql import expression as expression
 from ..sql import operators as operators
 from ..sql import roles as roles
@@ -45,6 +45,19 @@ def foreign(expr: Any): ...
 _T = TypeVar("_T")
 
 _InfoDict = Mapping[Any, Any]
+_OrderByArgument = Union[
+    Literal[False],
+    str,
+    elements.ColumnElement[Any],
+    Sequence[elements.ColumnElement[Any]],
+    Callable[
+        [],
+        Union[
+            elements.ColumnElement[Any],
+            Sequence[elements.ColumnElement[Any]],
+        ],
+    ],
+]
 
 class RelationshipProperty(StrategizedProperty[_T]):
     strategy_wildcard_key: str
@@ -91,9 +104,7 @@ class RelationshipProperty(StrategizedProperty[_T]):
         secondaryjoin: Optional[Any] = ...,
         foreign_keys: Optional[Any] = ...,
         uselist: Optional[bool] = ...,
-        order_by: Union[
-            Literal[False], str, Column, Callable[[], Column]
-        ] = ...,
+        order_by: _OrderByArgument = ...,
         backref: Union[str, _BackrefResult] = ...,
         back_populates: str = ...,
         overlaps: Union[AbstractSet[str], str] = ...,

--- a/test/files/relationship_order_by_ticket_77.py
+++ b/test/files/relationship_order_by_ticket_77.py
@@ -1,0 +1,62 @@
+from sqlalchemy import Column
+from sqlalchemy import create_engine
+from sqlalchemy import ForeignKey
+from sqlalchemy import Integer
+from sqlalchemy import select
+from sqlalchemy.orm import joinedload
+from sqlalchemy.orm import Mapped
+from sqlalchemy.orm import registry
+from sqlalchemy.orm import relationship
+from sqlalchemy.orm import Session
+
+mapper_registry: registry = registry()
+
+e = create_engine("sqlite:///")
+
+
+@mapper_registry.mapped
+class A:
+    __tablename__ = "a"
+    id = Column(Integer, primary_key=True)
+    b_id = Column(Integer, ForeignKey("b.id"))
+    number = Column(Integer, primary_key=True)
+    number2 = Column(Integer, primary_key=True)
+
+
+@mapper_registry.mapped
+class B:
+    __tablename__ = "b"
+    id = Column(Integer, primary_key=True)
+
+    # Omit order_by
+    a1: Mapped[A] = relationship("A", uselist=True)
+
+    # All kinds of order_by
+    a2: Mapped[A] = relationship("A", uselist=True, order_by=(A.id, A.number))
+    a3: Mapped[A] = relationship("A", uselist=True, order_by=[A.id, A.number])
+    a4: Mapped[A] = relationship("A", uselist=True, order_by=A.id)
+    a5: Mapped[A] = relationship("A", uselist=True, order_by=A.__table__.c.id)
+    a6: Mapped[A] = relationship("A", uselist=True, order_by="A.number")
+
+    # Same kinds but lambda'd
+    a7: Mapped[A] = relationship(
+        "A", uselist=True, order_by=lambda: (A.id, A.number)
+    )
+    a8: Mapped[A] = relationship(
+        "A", uselist=True, order_by=lambda: [A.id, A.number]
+    )
+    a9: Mapped[A] = relationship("A", uselist=True, order_by=lambda: A.id)
+
+
+mapper_registry.metadata.drop_all(e)
+mapper_registry.metadata.create_all(e)
+
+with Session(e) as s:
+    s.execute(select(B).options(joinedload(B.a1)))
+    s.execute(select(B).options(joinedload(B.a2)))
+    s.execute(select(B).options(joinedload(B.a3)))
+    s.execute(select(B).options(joinedload(B.a4)))
+    s.execute(select(B).options(joinedload(B.a5)))
+    s.execute(select(B).options(joinedload(B.a7)))
+    s.execute(select(B).options(joinedload(B.a8)))
+    s.execute(select(B).options(joinedload(B.a9)))

